apis/v1/gateway_types.go

```diff
@@ -788,6 +788,11 @@ type GatewayInfrastructure struct {
 	// the merging behavior is implementation specific.
 	// It is generally recommended that GatewayClass provides defaults that can be overridden by a Gateway.
 	//
+	// If the referent cannot be found, refers to an unsupported kind, or when
+	// the data within that resource is malformed, the Gateway SHOULD be
+	// rejected with the "Accepted" status condition set to "False" and an
+	// "InvalidParameters" reason.
```

Comment:
Seeing this late after reviewing the release notes.

So with this suggestion, there could be a case where a user has a proxy provisioned and running with a valid Gateway, then makes a change to the parametersRef resource that is invalid, and now the Gateway could be marked as invalid and the entire Route tree below it is nullified (and depending on the implementation, the proxy could be deprovisioned because the Gateway is no longer valid). That's pretty disruptive.